server/src/main/kotlin/com/kamelia/hedera/rest/user/Service.kt

```diff
@@ -154,17 +156,23 @@ private suspend fun checkEmail(email: String?, toEdit: User? = null) =
  *
  * @return Optional [Response] with [ErrorDTO] if error occurred
  */
-private suspend fun checkUsername(username: String?, toEdit: User? = null) =
-    if (username != null)
-        Users.findByUsername(username)
+private suspend fun checkUsername(username: String?, toEdit: User? = null): Response<Nothing, ErrorDTO>? {
+    if (username != null) {
+        if (!USERNAME_REGEX.matches(username)) {
+            return Response.badRequest(Errors.Users.Username.INVALID_USERNAME)
+        }
+
+        Users.findByUsername(username.lowercase())
```

Comment:
lowercase even less needed

Reply:
Honestly, now that I think about it, shouldn't it be better to let the ORM throw for a unicity constraint ? (instead of making a request to find a user, and then another one to insert it)

Reply:
You are kinda right but, first I feel like ORM exceptions are the last rampart to the uniqueness integrity, and second I don't know if we could easily catch the error to send the appropriate error code...

Reply:
https://github.com/Black-Kamelia/Hedera/issues/83#issuecomment-1584522678

i made the concession to restrict the username mainly for performance reasons. so for this time, I think we should actually go for my suggestion to avoid a request and let the db throw.

if you want, I'll look into how to catch the unicity constraint on the name.

Reply:
Ok, so I'm assigning you to the issue as well, and you will work on this part and the login form.